### PR TITLE
Migrate current dispatcher to use `ReactiveKafkaConsumer` instead of Vertx KafkaConsumer.

### PR DESCRIPTION
**Pull request by debasishbsws**
progress #2928 
The new dispatcher-vertx module is the one that uses Vertx implementation.

## Proposed Changes

- Remove all the Vertx Kafka client dependencies from the base dispatcher module.
- Change the current `main()` to `start(..)` and call from the other module with specialised Consumer implementation as args.
- migrate all the tests from Direct KafkaConsumer to use ReactiveKafkaConsumer Interface.

## Aditional Info

- The implementation of passing handler function to `ConsumerRebalanceListener` has gotten complex. I will simplify that further in the future.
- Tests for the dispatcher-vertx module are not there as vertx implementations are already well tested.

### CODE REVIEW

**Comment by debasishbsws** [comment]
@pierDipi In the last commit basically I create `ConsumerRebalanceListener` from the `PartisionsRevokeHandlers` and pass it to `ConsumerVirticleContext` and then while calling the `subscribe()` method I am passing that ConsumerRebalanceListener with it. There may by some redundancy now like in future I can change the PartisionsRevokeHandlers in order to reduce the complexity.

**Comment by debasishbsws** [comment]
Didn't look into the Benchmark module properly just migrate it to use Vertx implementation.

**Comment by debasishbsws** [comment]
I am not sure why the `profiler` test is failing.

**Comment by debasishbsws** [comment]
Though tests are saying to run `update-codegen.sh`, but after running that `DataPlaneContract.java`'s `hasReplyUrf()` method is being removed.
image

But we need this as we call it from
https://github.com/knative-sandbox/eventing-kafka-broker/blob/fb89db60df3fdb7d6ae6721037a081b5d88a60c8/data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/main/ConsumerVerticleBuilder.java#L204-L206

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/consumer/OrderedConsumerVerticle.java [hunk 1]
```diff
@@ -157,34 +160,35 @@ public void stop(Promise<Void> stopPromise) {
...
-      bucket.forceAddTokens(records.size());
+      bucket.forceAddTokens(records.count());
     }
 
     // Put records in internal per-partition queues.
-    for (int i = 0; i < records.size(); i++) {
-      final var record = records.recordAt(i);
+    var recordIterator = records.iterator();
+    while(recordIterator.hasNext()) {
+      var record = recordIterator.next();
```
[the remark this answers is not in the document]
Comment (debasishbsws) [reply data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/consumer/OrderedConsumerVerticle.java]: It's Done.

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**